core/src/main/scala/org/apache/spark/storage/StorageUtils.scala

```diff
@@ -60,11 +63,17 @@ class StorageStatus(val blockManagerId: BlockManagerId, val maxMem: Long) {
    * non-RDD blocks contains only the first 3 fields (in the same order).
    */
   private val _rddStorageInfo = new mutable.HashMap[Int, (Long, Long, StorageLevel)]
-  private var _nonRddStorageInfo: (Long, Long) = (0L, 0L)
+
+  // On-heap memory, off-heap memory and disk usage of non rdd storage
+  private var _nonRddStorageInfo: (Long, Long, Long) = (0L, 0L, 0L)
```

Comment:
Another preference, I'd rather see this as perhaps a case class so you could do something like:
_nonRddStorageInfo(onHeap: Long, offHeap: Long, disk: Long)

That way you don't need to do things like _1 and _2 below which, IMO, make the code less readable.